plugins/blueprint.cpp

```diff
@@ -592,179 +612,200 @@ command_result do_transform(DFCoord start, DFCoord end, string name, uint32_t ph
         return CR_FAILURE;
     }
 
-    if (phases & QUERY)
+    if (options.auto_phase || options.query)
```

Comment:
I may have missed something, but it looks to me like every check for `auto_phase` is of the form `auto_phase ||  some_mode`. Would it make more sense to just enable every mode if no mode is specified (or whatever triggers `auto_phase` currently)?

Reply:
this is a forward-looking design decision. once I change how the blueprint data is buffered and output, only blueprints that actually have data will be written. the user will be able to force creation of otherwise empty blueprints (e.g. for getting a pretty-printed blank spreadsheet for editing purposes) by explicitly requesting the phase. so for that I'll need the original values of both auto_phase and the flag for the phase itself.

Reply:
Makes sense - I almost asked about the absence of logic to handle that.